lib/opam_solve.ml

```diff
@@ -196,6 +196,86 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     | Error _ -> false
     | Ok opam_file -> Opam.has_cross_compile_tag opam_file
 
+  (* this is a hack - as we don't have a global view of the package
+     universe, we build the dev_repo table lazily, whenever the 0install
+     solver request more information about a package (by calling
+     [Solver.candidates]. *)
+  let dev_repos = ref Dev_repo.Map.empty
```

Comment:
Could also use a `Hashtbl` here? No strong opinions on this either way.

Reply:
updated with an hashtable